_delphi_utils_python/delphi_utils/covidcast_wrapper.py

```diff
@@ -0,0 +1,251 @@
+      available at the "week" resolution, representing an MMWR week ("epiweek").
+    :returns: A Pandas data frame with matching data, or ``None`` if no data is
+      returned. Each row is one observation on one day in one geographic location.
+      Contains the following columns:
+
+      ``geo_value``
+        Identifies the location, such as a state name or county FIPS code. The
+        geographic coding used by COVIDcast is described in the `API
+        documentation here
+        <https://cmu-delphi.github.io/delphi-epidata/api/covidcast_geography.html>`_.
+
+      ``signal``
+        Name of the signal, same as the value of the ``signal`` input argument. Used for
+        downstream functions to recognize where this signal is from.
+
+      ``time_value``
+        Contains a `pandas Timestamp object
+        <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timestamp.html>`_
+        identifying the date this estimate is for. For data with ``time_type = "week"``, this
+        is the first day of the corresponding epiweek.
+
+      ``issue``
+        Contains a `pandas Timestamp object
+        <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timestamp.html>`_
+        identifying the date this estimate was issued. For example, an estimate
+        with a ``time_value`` of June 3 might have been issued on June 5, after
+        the data for June 3rd was collected and ingested into the API.
+
+      ``lag``
+        Integer giving the difference between ``issue`` and ``time_value``,
+        in days.
+
+      ``value``
+        The signal quantity requested. For example, in a query for the
+        ``confirmed_cumulative_num`` signal from the ``usa-facts`` source,
+        this would be the cumulative number of confirmed cases in the area, as
+        of the ``time_value``.
+
+      ``stderr``
+        The value's standard error, if available.
+
+      ``sample_size``
+        Indicates the sample size available in that geography on that day;
+        sample size may not be available for all signals, due to privacy or
+        other constraints.
+
+      ``geo_type``
+        Geography type for the signal, same as the value of the ``geo_type`` input argument.
+        Used for downstream functions to parse ``geo_value`` correctly
+
+      ``data_source``
+        Name of the signal source, same as the value of the ``data_source`` input argument. Used for
+        downstream functions to recognize where this signal is from.
+
+    Consult the `signal documentation
+    <https://cmu-delphi.github.io/delphi-epidata/api/covidcast_signals.html>`_
+    for more details on how values and standard errors are calculated for
+    specific signals.
+
+    """
+    if start_day > end_day:
+        raise ValueError(
+            "end_day must be on or after start_day, but " f"start_day = '{start_day}', end_day = '{end_day}'"
+        )
+
+    time_values = list(date_generator(start_day, end_day))
```

Comment:
You can get rid of this generator for a few reasons: its not going to work properly for "week"-type data, and its gonna be inefficient...
```suggestion
    time_values = f"{start_day}-{end_day}"
```
I think this suggestion should work without exploding out into a big comma-separated list, which will be slightly gentler on the API server and probably also better for the DB optimizer.

Reply:
You could also try using the `Epidata.range()` method (like [this](https://github.com/cmu-delphi/delphi-epidata/blob/8746ff2ef7a936bb93628bc1358471d7c6c4f5f8/integrations/client/test_delphi_epidata.py#L152-L154)).  Its more verbosity and obfuscation, but its kinda nicer from an object-oriented perspective

Reply:
The reason I created a generator is that when I looked into the epidata.covidcast() call it was building a query to call the db and for the time values. I tried to use the something similar and it didn't work.

And I just tried using the Epidata.range and the result didn't work. tried setting the range with str and datetime.

```
from delphi_epidata import Epidata
startdate = datetime(year=2024, month=7, day=19)
enddate = datetime(year=2024, month=7, day=24)
sdate_str = startdate.strftime('%Y-%m-%d')
edate_str = enddate.strftime('%Y-%m-%d')

def date_generator(startdate, enddate):
  while startdate <= enddate:
    yield startdate.strftime('%Y-%m-%d')
    startdate = startdate + timedelta(days=1)

response = Epidata.covidcast(source, signal, time_type="day",
                          geo_type=geo, time_values=Epidata.range(sdate_str, edate_str),
                          geo_value="*", as_of=None, lag=None)
response2 = Epidata.covidcast(source, signal, time_type="day",
                          geo_type=geo, time_values=Epidata.range(startdate, enddate),
                          geo_value="*", as_of=None, lag=None)
response3 = Epidata.covidcast(source, signal, time_type="day",
                          geo_type=geo, time_values=list(date_generator(startdate, enddate)),
                          geo_value="*", as_of=None, lag=None)
```

Reply:
I realized that the Epidata.range was used specifically for issues, but I still need to get the data based on the time value and in epidata the time value needs to be filled out, and from the previous comment the epidata.range or string does not work.

in the example you showed the time_value was set to one date, where we would see rows where the time value is a specific date 2020_02_02 associated with multiple issue_dates.

So this does give me data back:
```
response6 = Epidata.covidcast(source, signal, time_type="day",
                          geo_type=geo,  time_values=sdate_str, issues=Epidata.range("20240629", "20240719"),
                          geo_value="*", as_of=None, lag=None)
```
but I would still need to call for multiple days unless I use the exploded comma dates

Reply:
The `time_values` param to the `/covidcast` can definitely handle time ranges, for instance

```
https://api.delphi.cmu.edu/epidata/covidcast/?data_source=fb-survey&signals=smoothed_cli&time_type=day&time_values=20210405-20210410&geo_type=state&geo_values=ca
```

 so there's something going wrong client side. May the string isn't formatted correctly?

Reply:
yup -- the strings aren't formatted correctly. Didn't realized that the string format for the epidata api were %Y%m%d

Reply:
If youre using the range notation properly, you shouldnt need the comma-separated list, and you should be able to make just one request.  You could try turning on the client's DEBUG mode to see what its actually requesting.